exhibitor-core/src/main/java/com/netflix/exhibitor/core/config/azure/AzureConfigProvider.java

```diff
@@ -0,0 +1,147 @@
+        Date lastModified;
+        Properties properties = new Properties();
+        CloudBlob object = getConfigObject();
+        if (object != null) {
+            lastModified = object.getProperties().getLastModified();
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            object.download(out);
+            properties.load(new ByteArrayInputStream(out.toByteArray()));
+        } else {
+            lastModified = new Date(0L);
+        }
+
+        PropertyBasedInstanceConfig config = new PropertyBasedInstanceConfig(properties, defaults);
+        return new LoadedInstanceConfig(config, lastModified.getTime());
+    }
+
+    @Override
+    public LoadedInstanceConfig storeConfig(ConfigCollection config, long compareVersion) throws Exception {
+        {
+            BlobProperties blobProperties = getConfigBlobProperties();
+            if (blobProperties != null) {
+                Date lastModified = blobProperties.getLastModified();
+                if (lastModified.getTime() != compareVersion) {
+                    return null;    // pattern copied from S3ConfigProvider - Azure may support a better way
+                }
+            }
+        }
+
+        PropertyBasedInstanceConfig propertyBasedInstanceConfig = new PropertyBasedInstanceConfig(config);
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        propertyBasedInstanceConfig.getProperties().store(out, "Auto-generated by Exhibitor " + hostname);
+
+        byte[] bytes = out.toByteArray();
+        azureClient.putBlob(bytes, arguments.getContainer(), arguments.getUri());
+
+        BlobProperties blobProperties = azureClient.getBlobProperties(arguments.getContainer(), arguments.getUri());
+        return new LoadedInstanceConfig(propertyBasedInstanceConfig, blobProperties.getLastModified().getTime());
+    }
+
+    private BlobProperties getConfigBlobProperties() throws Exception {
+        try {
+            BlobProperties properties = azureClient.getBlobProperties(arguments.getContainer(), arguments.getUri());
+            if (properties.getLength() > 0) {
+                return properties;
+            }
+        } catch (StorageException e) {
+            if (!isNotFoundError(e)) {
+                throw e;
+            }
+        }
+        return null;
+    }
+
+    private CloudBlob getConfigObject() throws Exception {
+        try {
+            CloudBlob object = azureClient.getBlob(arguments.getContainer(), arguments.getUri());
+            object.downloadAttributes();
+            if (object.getProperties().getLength() > 0) {
+                return object;
+            }
+        } catch (StorageException e) {
+            if (!isNotFoundError(e)) {
+                throw e;
+            }
+        }
+        return null;
+    }
+
+    private boolean isNotFoundError(StorageException e) {
+        return (e.getHttpStatusCode() == 404) || (e.getHttpStatusCode() == 403);
```

Comment:
argh! 403 is FORBIDDEN, not NOT FOUND: why are we muddling the waters here?
this should at least be well documented, as it's likely to confuse people (and silently hide bugs)

Also, please do not use hard-coded "magic numbers" (I know, they're unlikely to ever change, but still...) there are constants defined for all the HTTP codes: much *more* readable